p2p/base_reactor.go

```diff
@@ -86,24 +85,34 @@ func NewBaseReactor(name string, impl Reactor, opts ...ReactorOptions) *BaseReac
 		ctx:         ctx,
 		cancel:      cancel,
 		BaseService: *service.NewBaseService(nil, name, impl),
-		Switch:      nil,
+		Switch:      nil, // set by the switch later
 		incoming:    make(chan UnprocessedEnvelope, 100),
-		processor:   DefaultProcessor(impl),
+		processor:   nil, // Will be set after base is created
 	}
 	base.queueingFunc = base.QueueUnprocessedEnvelope
 	for _, opt := range opts {
 		opt(base)
 	}
 
+	// Set the processor after base is created, only if it hasn't been set by options
+	if base.processor == nil {
+		base.processor = ProcessorWithReactor(impl, base)
+	}
+
 	go func() {
-		err := base.processor(ctx, base.incoming)
-		if err != nil {
-			fmt.Println("processor exited with error ", err)
-			err = base.Stop()
-			if err != nil {
-				panic(err)
+		defer func() {
+			if r := recover(); r != nil {
+				fmt.Printf("processor panicked: %v\n", r)
```

Comment:
Would it be better to use the logger. Node operators might not be monitoring std out